internal/host/plugin/repository_host_catalog_test.go

```diff
@@ -1441,6 +1443,162 @@ func TestRepository_DeleteCatalogX(t *testing.T) {
+
+	numSetsUpdated, err = dbRW.Update(ctx, set2, []string{"LastSyncTime", "NeedSync"}, []string{})
+	require.NoError(t, err)
+	require.Equal(t, 1, numSetsUpdated)
+
+	jobRunCh := make(chan struct{})
+	j := &testSyncJob{
+		F: func() { close(jobRunCh) },
+	}
+	err = sched.RegisterJob(ctx, j, scheduler.WithNextRunIn(setSyncJobRunInterval*2))
+	require.NoError(t, err)
+	var wg sync.WaitGroup
+	err = sched.Start(ctx, &wg)
+	require.NoError(t, err)
+
+	repo, err := NewRepository(dbRW, dbRW, dbKmsCache, sched, dummyPluginMap)
+	require.NoError(t, err)
+	require.NotNil(t, repo)
+
+	// Updating an empty catalog should not trigger an update.
+	emptyTestCatalog.Attributes = mustMarshal(map[string]interface{}{"foo": "bar"})
+	var catalogsUpdated int
+	_, _, catalogsUpdated, err = repo.UpdateCatalog(ctx, emptyTestCatalog, emptyTestCatalog.Version, []string{"attributes"})
+	require.NoError(t, err)
+	require.Equal(t, 1, catalogsUpdated)
+
+	// Wait 2m for the job to not run. The regular sync job is 10
+	// minutes, and we set the "next run" for the test job above to
+	// double that, so waiting 2 minutes should be fine, with the
+	// default job interval being 1m.
+	var jobRan bool
+	select {
+	case <-time.After(time.Minute * 2):
+	case <-jobRunCh:
+		jobRan = true
+	}
+
+	require.False(t, jobRan)
+
+	// Updating name should not flag sets for sync
+	testCatalog.Name = "foobar"
+	testCatalog, _, catalogsUpdated, err = repo.UpdateCatalog(ctx, testCatalog, testCatalog.Version, []string{"name"})
+	require.NoError(t, err)
+	require.Equal(t, 1, catalogsUpdated)
+
+	gotSet1, _, err := repo.LookupSet(ctx, set1.PublicId)
+	require.NoError(t, err)
+	require.False(t, gotSet1.NeedSync)
+	require.Equal(t, set1.LastSyncTime, gotSet1.LastSyncTime)
+	require.Equal(t, set1.Version, gotSet1.Version)
+
+	gotSet2, _, err := repo.LookupSet(ctx, set2.PublicId)
+	require.NoError(t, err)
+	require.False(t, gotSet2.NeedSync)
+	require.Equal(t, set2.LastSyncTime, gotSet2.LastSyncTime)
+	require.Equal(t, set2.Version, gotSet2.Version)
+
+	// Wait 2m again for the job to not run.
+	select {
+	case <-time.After(time.Minute * 2):
+	case <-jobRunCh:
+		jobRan = true
+	}
+
+	require.False(t, jobRan)
+
+	// Updating attributes should trigger update
+	testCatalog.Attributes = mustMarshal(map[string]interface{}{"foo": "bar"})
+	_, _, catalogsUpdated, err = repo.UpdateCatalog(ctx, testCatalog, testCatalog.Version, []string{"attributes"})
+	require.NoError(t, err)
+	require.Equal(t, 1, catalogsUpdated)
+
+	gotSet1, _, err = repo.LookupSet(ctx, set1.PublicId)
+	require.NoError(t, err)
+	require.True(t, gotSet1.NeedSync)
+	require.Equal(t, set1.Version+1, gotSet1.Version)
+
+	gotSet2, _, err = repo.LookupSet(ctx, set2.PublicId)
+	require.NoError(t, err)
+	require.True(t, gotSet2.NeedSync)
+	require.Equal(t, set2.Version+1, gotSet2.Version)
+
+	// Wait for the job again, for the same amount of time. This time the job should have ran.
+	select {
+	case <-time.After(time.Minute * 2):
```

Comment:
Having these potentially super long running tests aren't great.  The scheduler is tested separately so if it gets called with the expected input it should function properly.  Is there a way we can mock the scheduler or fake it in a way that we just have to confirm that it is called iunstead of waiting for up to 2 minute?

Beyond making sure the scheduler is called with scheduling the next job run soon we can try running the job using `TestRunSetSync` and see the sets are picked up.

Reply:
I don't think mocking it is an option, unfortunately, unless we change the repo over to take a full-on interface, which seems excessive for this purpose.

Let me see if I can whip something up so that we can check the next run time and assert that it's been set to 0 or not.

Reply:
Updated!